private/bufpkg/bufplugin/commit.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2020-2024 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bufplugin
+
+import (
+	"sync"
+	"time"
+)
+
+// Commit represents a Commit for a Plugin on the BSR.
+type Commit interface {
+	// PluginKey returns the PluginKey for the Commit.
+	PluginKey() PluginKey
+	// CreateTime returns the time the Commit was created on the BSR.
+	CreateTime() (time.Time, error)
+
+	isCommit()
+}
+
+// NewCommit returns a new Commit.
+func NewCommit(
+	pluginKey PluginKey,
+	getCreateTime func() (time.Time, error),
+) Commit {
+	return newCommit(
+		pluginKey,
+		getCreateTime,
+	)
+}
```

Comment:
Similar to digest, it might make sense to split `Commit` into its own package as well. Same question/concern around whether that should be a follow-up PR.